pkg/utils/net/ipnet_test.go

```diff
@@ -101,3 +101,44 @@ var _ = DescribeTable("#GetByIPFamily",
 	Entry("should return one entry with two IPv4 entries + IPv6 entry + ipv6 match", []string{"14.0.0.0/8", "15.0.0.0/8", "2001:db8:1::/64"}, IPv6Family, HaveLen(1)),
 	Entry("should return two entries with two IPv4 entries + two IPv6 entries + ipv6 match", []string{"16.0.0.0/8", "17.0.0.0/8", "2001:db8:1::/64", "2001:db8:2::/64"}, IPv6Family, HaveLen(2)),
 )
+
+var _ = DescribeTable("#Overlap",
+	func(cidr1, cidr2 string, expected bool) {
+		_, ipNet1, err := net.ParseCIDR(cidr1)
+		Expect(err).ToNot(HaveOccurred())
+		_, ipNet2, err := net.ParseCIDR(cidr2)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(Overlap(*ipNet1, *ipNet2)).To(Equal(expected))
+	},
+
+	Entry("should detect IPv4 overlap - same network", "10.0.0.0/8", "10.0.0.0/8", true),
+	Entry("should detect IPv4 overlap - subset network", "10.0.0.0/8", "10.0.0.0/16", true),
+	Entry("should detect IPv4 overlap - superset network", "10.0.0.0/16", "10.0.0.0/8", true),
+	Entry("should not detect IPv4 overlap - different networks", "10.0.0.0/8", "172.16.0.0/12", false),
+	Entry("should detect IPv6 overlap - same network", "2001:db8::/32", "2001:db8::/32", true),
+	Entry("should detect IPv6 overlap - subset network", "2001:db8::/32", "2001:db8:1::/48", true),
+	Entry("should detect IPv6 overlap - superset network", "2001:db8:1::/48", "2001:db8::/32", true),
+	Entry("should not detect IPv6 overlap - different networks", "2001:db8::/32", "2001:db9::/32", false),
```

Comment:
All matches start with exactly the same IP. Do you want to change/add tests so that we also check something else?